Makefile

```diff
@@ -63,6 +63,7 @@ vet: ## Run go vet against code.
 .PHONY: test
 test: manifests generate fmt vet envtest checklicense ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go run github.com/onsi/ginkgo/v2/ginkgo --label-filter="!integration" -coverprofile cover.out ./...
+	go mod tidy
```

Comment:
do we need this command?

Reply:
Yes. As it cleans test related dependencies from go.sum file after tests are run. This is added in other repositories as well like `gardener-extension-provider-onmetal`.

Reply:
I think we should remove it since it seems to be a temporary fix. The pipeline also checks if the go.mod is up to-date.

Reply:
done